cmd/observe/flows_filter.go

```diff
@@ -546,6 +547,15 @@ func (of *flowFilter) set(f *filterTracker, name, val string, track bool) error
 			f.HttpUrl = append(f.GetHttpUrl(), val)
 		})
 
+	case "http-header":
+		f.apply(func(f *flowpb.FlowFilter) {
+			key, hVal, found := strings.Cut(val, ":")
+			if found {
+				header := &flowpb.HTTPHeader{Key: key, Value: hVal}
+				f.HttpHeader = append(f.GetHttpHeader(), header)
+			}
```

Comment:
Sorry, one more thing that just came to mind now. I think we should return an error in case no colon was found, i.e. an invalid HTTP header:

```suggestion
			if !found {
				return fmt.Errorf("invalid http-header value %q, expected name:value", val)
			}
			header := &flowpb.HTTPHeader{Key: key, Value: hVal}
			f.HttpHeader = append(f.GetHttpHeader(), header)

```

Would be nice extend `TestHTTPHeader` accordingly as well.

Reply:
Sure! I have added that.